benchmarks/triangulated_surface_distance/triangulated_surface_distance.cpp

```diff
@@ -0,0 +1,303 @@
+      if (it != hash.end())
+      {
+        vmid[j] = it->second;
+      }
+      else
+      {
+        vmid[j] = vindex;
+        hash[std::make_pair(vmin, vmax)] = vindex;
+        ++vindex;
+
+        vertices.push_back(Point{(vertices[vmin][0] + vertices[vmax][0]) / 2,
+                                 (vertices[vmin][1] + vertices[vmax][1]) / 2,
+                                 (vertices[vmin][2] + vertices[vmax][2]) / 2});
+      }
+    }
+
+    new_triangles.insert(new_triangles.end(), {v[0], vmid[0], vmid[2]});
+    new_triangles.insert(new_triangles.end(), {v[1], vmid[0], vmid[1]});
+    new_triangles.insert(new_triangles.end(), {v[2], vmid[2], vmid[1]});
+    new_triangles.insert(new_triangles.end(), {vmid[0], vmid[1], vmid[2]});
+  }
+  triangles = new_triangles;
+}
+
+void projectVerticesToSphere(std::vector<Point> &vertices, float radius)
+{
+  for (auto &v : vertices)
+  {
+    auto norm = std::sqrt(v[0] * v[0] + v[1] * v[1] + v[2] * v[2]);
+    v[0] *= radius / norm;
+    v[1] *= radius / norm;
+    v[2] *= radius / norm;
+  }
+}
+
+void buildTriangles(float radius, int num_refinements,
+                    std::vector<Point> &vertices, std::vector<int> &triangles)
+{
+  Kokkos::Profiling::ScopedRegion guard("Benchmark::build_triangles");
+
+  icosahedron(vertices, triangles);
+  for (int i = 1; i <= num_refinements; ++i)
+    subdivide(vertices, triangles);
+
+  projectVerticesToSphere(vertices, radius);
+}
+
+template <typename... P, typename T>
+auto vec2view(std::vector<T> const &in, std::string const &label = "")
+{
+  Kokkos::View<T *, P...> out(
+      Kokkos::view_alloc(label, Kokkos::WithoutInitializing), in.size());
+  Kokkos::deep_copy(out, Kokkos::View<T const *, Kokkos::HostSpace,
+                                      Kokkos::MemoryTraits<Kokkos::Unmanaged>>{
+                             in.data(), in.size()});
+  return out;
+}
+
+template <typename MemorySpace>
+struct Triangles
+{
+  Kokkos::View<Point *, MemorySpace> _points;
+  Kokkos::View<int *, MemorySpace> _triangle_vertices;
+};
+
+template <typename MemorySpace>
+class ArborX::AccessTraits<Triangles<MemorySpace>, ArborX::PrimitivesTag>
+{
+  using Self = Triangles<MemorySpace>;
+
+public:
+  using memory_space = MemorySpace;
+
+  static KOKKOS_FUNCTION auto size(Self const &self)
+  {
+    return self._triangle_vertices.size() / 3;
+  }
+  static KOKKOS_FUNCTION auto get(Self const &self, int i)
+  {
+    auto const &vertices = self._triangle_vertices;
+    return Triangle{self._points(vertices(3 * i + 0)),
+                    self._points(vertices(3 * i + 1)),
+                    self._points(vertices(3 * i + 2))};
+  }
+};
+
+struct DistanceCallback
+{
+  template <typename Predicate, typename Value, typename OutputFunctor>
+  KOKKOS_FUNCTION void operator()(Predicate const &predicate,
+                                  Value const &value,
+                                  OutputFunctor const &out) const
+  {
+    using ArborX::Details::distance;
```

Comment:
OK since we are not in examples/ but wondering whether we need to do something about it.

Reply:
Like what? We can simply remove it, I guess, as we control the geometries in the benchmark. But if a user looks here at any point, we may want to leave this around.

Reply:
Add a comment, discuss whether we want to officially support geometric functions, etc.

Reply:
> Add a comment, discuss whether we want to officially support geometric functions, etc.

I don't understand what you are saying.

Reply:
You are right. We don't need it here.